### PR TITLE
[refactor](pipelineX) refine _build_side_pipelines

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

File: be/src/pipeline/pipeline_x/pipeline_x_fragment_context.h [hunk 1]
```diff
@@ -166,14 +166,29 @@ class PipelineXFragmentContext : public PipelineFragmentContext {
     // build probe operator and build operator in separate pipelines. To do this, we should build
     // ProbeSide first, and use `_pipelines_to_build` to store which pipeline the build operator
     // is in, so we can build BuildSide once we complete probe side.
-    std::map<int, PipelinePtr> _build_side_pipelines;
+    struct Pipelines_Stack {
```
Comment (Gabriel39) [inline be/src/pipeline/pipeline_x/pipeline_x_fragment_context.h:169]: It is not a real stack. Using other struct name will be more clear

**Comment by Mryange** [comment]
run buildall

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**